synapse/server_notices/consent_server_notices.py

```diff
@@ -81,10 +86,18 @@ def maybe_send_server_notice_to_user(self, user_id):
                 # we've already sent a notice to the user
                 return
 
-            # need to send a message
+            # need to send a message.
             try:
+                consent_uri = self._consent_uri_builder.build_user_consent_uri(
+                    get_localpart_from_id(user_id),
+                )
+                content = copy_with_str_subst(
+                    self._server_notice_content, {
+                        'consent_uri': consent_uri,
+                    },
+                )
```

Comment:
I really don't like that the idea of magically going and rewriting everything like this, but am prepared to accept it for now since presumably we need the flexibility.